### PR TITLE
feat: Added visionOS support for Apple Vision Pro

### PR DESCRIPTION
**Pull request by eugenehp**
Saw good work in https://github.com/rust-lang/socket2/pull/503 

This PR addresses the comments with `target_os` specifics.

Tested it with [Tier 3](https://github.com/rust-lang/rust/pull/121419) target.

### CODE REVIEW

**Comment by eugenehp** [comment]
CI pulls old nightly from 2018, should pull the recent one instead,

Screenshot 2024-06-04 at 10 29 23 AM

**Comment by eugenehp** [comment]
Looks like CI for freebsd is broken
`The resource 'projects/freebsd-org-cloud-dev/global/images/family/freebsd-13-2' was not found`

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        target: ["aarch64-apple-ios", "aarch64-linux-android", "x86_64-apple-darwin", "x86_64-unknown-fuchsia", "x86_64-pc-windows-msvc", "x86_64-pc-solaris", "x86_64-unknown-freebsd", "x86_64-unknown-illumos", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl", "x86_64-unknown-netbsd", "x86_64-unknown-redox", "armv7-linux-androideabi", "i686-linux-android"]
```
Comment (Thomasdezeeuw) [inline .github/workflows/main.yml:59]: No. That's not happening.

Why would you delete an OS in a pr to add another?

**Review by Thomasdezeeuw — COMMENTED** [review]
This pr is not ok.

You mention #503, but then proceed to not give the author credit.

You also remove another OS from the CI?

I'm sorry, but this is just unacceptable.

Closing in favour of #503.

File: .github/workflows/main.yml (hunk 1 above)
Comment (eugenehp) [reply .github/workflows/main.yml:59]: https://github.com/rust-lang/socket2/pull/514#issuecomment-2148043695

**Comment by eugenehp** [comment]
> This pr is not ok.
> 
> You mention #503, but then proceed to not give the author credit.

This code was done without prior knowledge of #503

> You also remove another OS from the CI?

https://github.com/rust-lang/socket2/pull/514#issuecomment-2148043695

> I'm sorry, but this is just unacceptable.

Ok, let me know if I can be of help with #503. Thanks.

> Closing in favour of #503.

**Comment by Thomasdezeeuw** [comment]
> > You also remove another OS from the CI?
> 
> [#514 (comment)](https://github.com/rust-lang/socket2/pull/514#issuecomment-2148043695)

The solution is not to remove another OS, especially not within a pr that adds another.

Why not communicate with the maintainers, i.e. me, on how to solve it?

> > I'm sorry, but this is just unacceptable.
> 
> Ok, let me know if I can be of help with #503. Thanks.

You can leave this patch on that pr so we can further discuss adding visionOS support.

**Pull request closed without merging**